File: app/src/main/java/vn/edu/usth/weather/WeatherPagerAdapter.java
package vn.edu.usth.weather;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import vn.edu.usth.weather.WeatherAndForecastFragment;

class WeatherPagerAdapter extends FragmentStateAdapter {
    public WeatherPagerAdapter(FragmentActivity fa) { super(fa); }

    @Override
    public Fragment createFragment(int position) {
        return new WeatherAndForecastFragment();
    }

    @Override
    public int getItemCount() { return 3; }
}

File: app/src/main/kotlin/vn/edu/usth/weather/WeatherPagerAdapter.kt
package vn.edu.usth.weather

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter

import vn.edu.usth.weather.WeatherAndForecastFragment

class WeatherPagerAdapter(fa: FragmentActivity) : FragmentStateAdapter(fa) {
    override fun getItemCount(): Int = 3
    override fun createFragment(position: Int): Fragment =
        WeatherAndForecastFragment()
}
